powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/util/ProtocolUtils.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2019 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.crypto.lib.util;
+
+import io.getlime.security.powerauth.crypto.lib.config.PowerAuthConfiguration;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.model.exception.GenericCryptoException;
+import io.getlime.security.powerauth.provider.exception.CryptoProviderException;
+
+import javax.crypto.SecretKey;
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+
+/**
+ * The {@code ProtocolUtils} class provides various additional functionality defined for PowerAuth protocol.
+ */
+public class ProtocolUtils {
```

Comment:
I was trying to find a right name, or put the method to some existing class, but didn't find a better solution. `InitializationVectorDerivationGenerator` will definitely satisfy your Java naming obsession, but sounds really strange :) 

To be honest, if I compare `KeyGenerator`﹣which contains a lot of completely unrelated methods, against `ProtocolUtils` ﹣which suppose to contain various methods related to PowerAuth protocol, then the 2nd option sounds better for me. Just because something's has "key" in its name that doesn't mean that it belong to one class.

Reply:
Would `KeyDerivationUtils` be a bad name? Your sample class name is too specific. Yes, naming classes can be hard :)